### PR TITLE
feat: Deferred start animation

### PR DESCRIPTION
**Pull request by Rhuagh**
Depends on #641.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/644)

### CODE REVIEW

File: examples/animation/main.rs [hunk 1]
```diff
@@ -13,30 +13,52 @@ use amethyst::renderer::{AmbientColor, Camera, DisplayConfig, DrawShaded, Elemen
...
+                         Sampler, SamplerPrimitive, StepDirection, TransformChannel};
 use genmesh::{MapToVertices, Triangulate, Vertices};
 use genmesh::generators::SphereUV;
 
-const SPHERE_COLOUR: [f32; 4] = [0.0, 0.0, 1.0, 1.0]; // blue
-const AMBIENT_LIGHT_COLOUR: Rgba = Rgba(0.01, 0.01, 0.01, 1.0); // near-black
-const POINT_LIGHT_COLOUR: Rgba = Rgba(1.0, 1.0, 1.0, 1.0); // white
-const BACKGROUND_COLOUR: [f32; 4] = [0.0, 0.0, 0.0, 0.0]; // black
+const SPHERE_COLOUR: [f32; 4] = [0.0, 0.0, 1.0, 1.0];
+// blue
```
Comment (jojolepro) [inline examples/animation/main.rs]: Comments should be on top, not after

File: examples/animation/main.rs [hunk 2]
```diff
@@ -63,30 +85,145 @@ impl State for Example {
...
+                                    &animation,
+                                    EndControl::Normal,
+                                    self.rate,
+                                    AnimationCommand::Start,
+                                    AnimationId::Translate,
+                                    DeferStartRelation::End,
+                                );
+                            }
+                            {
+                                let animation = world
```
Comment (jojolepro) [inline examples/animation/main.rs]: Any way to reuse this? The code duplication is getting out of control.

File: examples/animation/main.rs [hunk 3]
```diff
@@ -243,23 +380,46 @@ fn initialise_animation(world: &mut World) -> Handle<Animation<Transform>> {
...
-                translation_animation_handle,
-            ),
-            //(0, scale_animation_handle),
-            (0, TransformChannel::Rotation, rotation_animation_handle),
-        ],
-    };
-    loader.load_from_data(animation, (), &world.read_resource())
+    let animation_storage = world.read_resource();
+    let mut set = AnimationSet::<AnimationId, Transform>::new();
+    set.insert(
```
Comment (jojolepro) [inline examples/animation/main.rs]: Can you make a function please?

File: examples/animation/main.rs (hunk 1 above)
Comment (Rhuagh) [reply examples/animation/main.rs]: rustfmt ftl

Comment (Rhuagh) [reply examples/animation/main.rs]: they used to be after the line, it moved it down ..

File: examples/animation/main.rs (hunk 3 above)
Comment (Rhuagh) [reply examples/animation/main.rs]: Not sure it made it better, but there you go.

File: examples/animation/main.rs (hunk 2 above)
Comment (Rhuagh) [reply examples/animation/main.rs]: Done.

File: examples/animation/main.rs (hunk 1 above)
Comment (Rhuagh) [reply examples/animation/main.rs]: Done

**Pull request merged**